lib/icingadb/icingadb-objects.cpp

```diff
@@ -2177,6 +2275,159 @@ void IcingaDB::SendAcknowledgementCleared(const Checkable::Ptr& checkable, const
+
+		for (const User::Ptr& user : userGroup->GetMembers()) {;
+			String userId = HashValue(new Array(Prepend(GetEnvironment(), Prepend("user", Prepend(GetObjectIdentifiersWithoutEnv(user), GetObjectIdentifiersWithoutEnv(notification))))));
+			DeleteRelationship(userId, "notification:recipient");
+		}
+	}
+}
+
+void IcingaDB::SendTimePeriodRangesChanged(const TimePeriod::Ptr& timeperiod, const Dictionary::Ptr& oldValues, const Dictionary::Ptr& newValues) {
+	if (!m_Rcon || !m_Rcon->IsConnected() || oldValues == newValues) {
+		return;
+	}
+
+	std::vector<String> deletedKeys = GetDictionaryDeletedKeys(oldValues, newValues);
+	String typeName = GetLowerCaseTypeNameDB(timeperiod);
+
+	for (const auto& rangeKey : deletedKeys) {
+		String id = HashValue(new Array(Prepend(GetEnvironment(), Prepend(rangeKey, Prepend(oldValues->Get(rangeKey), GetObjectIdentifiersWithoutEnv(timeperiod))))));
+		DeleteRelationship(id, "timeperiod:range");
+	}
+}
+
+void IcingaDB::SendTimePeriodIncludesChanged(const TimePeriod::Ptr& timeperiod, const Array::Ptr& oldValues, const Array::Ptr& newValues) {
+	if (!m_Rcon || !m_Rcon->IsConnected() || oldValues == newValues) {
+		return;
+	}
+
+	std::vector<Value> deletedIncludes = GetArrayDeletedValues(oldValues, newValues);
+
+	for (const auto& includeName : deletedIncludes) {
+		TimePeriod::Ptr includeTp = TimePeriod::GetByName(includeName);
+		String id = HashValue(new Array(Prepend(GetEnvironment(), Prepend(GetObjectIdentifiersWithoutEnv(includeTp), GetObjectIdentifiersWithoutEnv(timeperiod)))));
+		DeleteRelationship(id, "timeperiod:override:include");
+	}
+}
+
+void IcingaDB::SendTimePeriodExcludesChanged(const TimePeriod::Ptr& timeperiod, const Array::Ptr& oldValues, const Array::Ptr& newValues) {
+	if (!m_Rcon || !m_Rcon->IsConnected() || oldValues == newValues) {
+		return;
+	}
+
+	std::vector<Value> deletedExcludes = GetArrayDeletedValues(oldValues, newValues);
+
+	for (const auto& excludeName : deletedExcludes) {
+		TimePeriod::Ptr excludeTp = TimePeriod::GetByName(excludeName);
+		String id = HashValue(new Array(Prepend(GetEnvironment(), Prepend(GetObjectIdentifiersWithoutEnv(excludeTp), GetObjectIdentifiersWithoutEnv(timeperiod)))));
+		DeleteRelationship(id, "timeperiod:override:exclude");
+	}
+}
+
+void IcingaDB::SendGroupsChanged(const ConfigObject::Ptr& object, const Array::Ptr& oldValues, const Array::Ptr& newValues) {
+	if (!m_Rcon || !m_Rcon->IsConnected() || oldValues == newValues) {
+		return;
+	}
+
+	std::vector<Value> deletedGroups = GetArrayDeletedValues(oldValues, newValues);
+	String typeName = GetLowerCaseTypeNameDB(object);
+	ConfigObject::Ptr (*getGroup)(const String& name);
+
+	if (typeName == "user") {
+		getGroup = &::GetObjectByName<UserGroup>;
+	} else if (typeName == "host") {
+		getGroup = &::GetObjectByName<HostGroup>;
+	} else {
+		getGroup = &::GetObjectByName<ServiceGroup>;
+	}
+
+	for (const auto& groupName : deletedGroups) {
+		auto group ((*getGroup)(groupName));
```

Comment:
Probably just (partially) copied over but I'm not a fan of C-style function pointers. Maybe at least use `std::function<...>` as type of `getGroup` or maybe templating the function could be an option?